src/core/prototypes/marks/data_axis.ts

```diff
@@ -371,6 +372,26 @@ export class DataAxisClass extends MarkClass<
       table: this.getGlyphClass().object.table,
       groupBy: null // TODO: fixme
     };
+    let properties: Specification.Template.Property[] = [];
+    if (this.object.properties.axis) {
+      const axis = this.object.properties.axis;
+      properties = properties.concat(
+        buildAxisProperties(this.object as any, "axis")
+      );
+    }
+    if (this.object.properties.axis) {
```

Comment:
It seems like you can combine there two `if(this.object.properties.axis)` statements

Reply:
Fixed